File: core/src/main/java/org/robotframework/roc/core/models/Project.java
package org.robotframework.roc.core.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.redis.core.RedisHash;

import java.io.Serializable;
import java.util.Date;

@Data
@NoArgsConstructor
@RedisHash("Project")
public class Project implements Serializable {

    private Long id;

    private String name;
    private String description;

    private Date createdAt;
    private Date updatedAt;

}

File: web/src/services/project.ts
import http from "./http"
import {Project} from "../models/Project";

export function fetchProjects() {
    return http.get<Array<Project>>("/projects");
}

export function fetchProjectById(id: string) {
    return http.get<Project>(`/projects/${id}`);
}

export default {
    fetchProjects: fetchProjects,
    fetchProjectById: fetchProjectById
}
